server/src/main/java/org/elasticsearch/health/metadata/HealthMetadata.java

```diff
@@ -142,6 +174,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field(FLOOD_STAGE_WATERMARK_FIELD.getPreferredName(), describeFloodStageWatermark());
             builder.field(FROZEN_FLOOD_STAGE_WATERMARK_FIELD.getPreferredName(), describeFrozenFloodStageWatermark());
             builder.field(FROZEN_FLOOD_STAGE_MAX_HEADROOM_FIELD.getPreferredName(), frozenFloodStageMaxHeadroom);
+            builder.field(HIGH_MAX_HEADROOM_FIELD.getPreferredName(), highMaxHeadroom);
```

Comment:
@gmarouli since I still see there is a `toXContent` function, I extend it with the max headroom fields. However, am I correct to understand that by converting the HealthMetadata from Metadata.Custom to ClusterState.Custom, there won't be ever a need to convert HealthMetadata to/from XContent? So I do not need to check for the version here to handle any bwc issue. Right? Furthermore, would it make sense to empty this function or throw a "Not implemented" exception or something since if it is never used?

Reply:
The `toXContent` function is still used in some APIs, e.g. `GET _cluster/state`, so I think it's best to keep it and add these fields here. It doesn't need to round-trip faithfully through `XContent` any more, but the more detail we see here the easier it is to debug problems in production.

Reply:
Thanks for the explanation! Resolving this conversation then.